xcp/compat.py

```diff
@@ -0,0 +1,59 @@
+"""Helper module for setting up binary or UTF-8 I/O for Popen and open in Python 3.6 and newer"""
```

Comment:
Are these from the xcp-python-libs repo? Perhaps we could use a github checkout action to bring in the right stuff for the CI's purpose (can you run a checkout action with a different repo into a subdir?).
Actually having these files present in XAPI might lead to some code duplication.

Reply:
Yes, a GitHub checkout from python-libs might be needed,
We'll see how to provide the type-information later in the PR that would do it.